ui/src/app/views/workspace/component.spec.ts

```diff
@@ -0,0 +1,109 @@
+import {Component, DebugElement} from '@angular/core';
+import {TestBed, async, fakeAsync, ComponentFixture} from '@angular/core/testing';
+import {Title, By} from '@angular/platform-browser';
+import {ActivatedRoute, UrlSegment} from '@angular/router';
+import {RouterTestingModule} from '@angular/router/testing';
+import {WorkspaceComponent} from 'app/views/workspace/component';
+import {updateAndTick, simulateInput} from 'testing/test-helpers';
+import {CohortsServiceStub} from 'testing/stubs/cohort-service-stub';
+import {CohortsService} from 'generated';
+import {UserService} from 'app/services/user.service';
+import {RepositoryService} from 'app/services/repository.service';
+
+class Context {
```

Comment:
I know we just set up the Context pattern, but now I'm wondering if it's right. Was there a reason we couldn't just merge it into the test class, and put its constructor into beforeEach? I think there was some question about what the async behavior in beforeEach did, but from what I've read it just means "finish all the async stuff in here", not any magic parallelization. Since every test case creates a Context in the same way, and different test classes are ending up with different contexts, I don't think splitting it out buys much (you can't read/edit the tests without having to understand the Context).